### PR TITLE
Add accuracy issue support in AOTI Minifier

### PR DESCRIPTION
**Pull request by yushangdi**
Summary:

Add three more repro levels for AOTI minifier (level 2 already exists). They are the same as the existing dynamo minifier repro levels.

Now AOTI minifier can minify and repro programs that have numerical accuracy issues as well.

1: Dumps the original graph out to repro.py if compilation fails
2: Dumps a minifier_launcher.py if aoti fails.
3: Always dumps a minifier_launcher.py. Good for segfaults.
4: Dumps a minifier_launcher.py if the accuracy fails.

Refactor AOTI minifier unit tests to be cleaner and better re-use the existing minifier testing code. We do not need to manually patch {"aot_inductor.dump_aoti_minifier": True} to each test now, this config is generated in the test code.

Differential Revision: D68294638

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @aakhundov

### CODE REVIEW

File: torch/_inductor/debug.py [hunk 1]
```diff
@@ -739,6 +743,37 @@ def aot_inductor_minifier_wrapper(
...
+            flat_example_inputs, config_copy = _aoti_flatten_inputs(
+                gm_copy,
+                example_inputs_copy[0],
+                example_inputs_copy[1],
+                options=config_copy,
+            )
+            tuple_inputs = tuple(flat_example_inputs)
+            flattened_ep = torch.export.export(gm_copy, tuple_inputs, strict=False)
+            func(
+                flattened_ep.module(),
```
Comment (desertfire) [inline torch/_inductor/debug.py:769]: Does this imply we are running the exported module in the eager model and use that as the baseline? If yes, then we are relying on export being correct. At some point, we may want to have a minifier for export itself.

**Review by desertfire — APPROVED** [review]
A general suggestion is to break refactoring into separate PR(s) next time for easier review.

File: torch/_inductor/debug.py (hunk 1 above)
Comment (yushangdi) [reply torch/_inductor/debug.py:769]: > Does this imply we are running the exported module in the eager mode and use that as the baseline? If yes, then we are relying on export being correct. At some point, we may want to have a minifier for export itself.

that's right, we are relying on export being correct. If we want to be extra careful, I can check against the result of the original unflattened eager model result as well in a follow up PR.

Have a minifier for export is probably do-able and not too hard if the input is a graph module (but not jus an nn.Module without an fx.Graph)
